File: src/main/java/hu/frontrider/gearcraft/util/OreDictUtil.java
package hu.frontrider.gearcraft.util;

import net.minecraft.block.Block;
import net.minecraft.item.Item;
import net.minecraft.item.ItemBlock;
import net.minecraft.item.ItemStack;
import net.minecraft.util.NonNullList;
import net.minecraftforge.oredict.OreDictionary;

public class OreDictUtil {

    public static boolean blockMatchesName(Block block, String name) {
        final NonNullList<ItemStack> oredictEntries = OreDictionary.getOres(name);
        boolean foundMatch = false;
        for (ItemStack itemStack : oredictEntries) {
            if (foundMatch)
                break;
            final Item item = itemStack.getItem();
            if (item instanceof ItemBlock) {
                if (((ItemBlock) item).getBlock() == block)
                    foundMatch = true;
            }
        }
        return foundMatch;
    }

}

File: mod/src/main/java/hu/frontrider/gearcraft/core/util/OreDictUtil.kt
package hu.frontrider.gearcraft.core.util

import net.minecraft.block.Block
import net.minecraft.item.Item
import net.minecraft.item.ItemBlock
import net.minecraft.item.ItemStack
import net.minecraft.util.NonNullList
import net.minecraftforge.oredict.OreDictionary

object OreDictUtil {

    fun blockMatchesName(block: Block, name: String): Boolean {
        val oredictEntries = OreDictionary.getOres(name)
        var foundMatch = false
        for (itemStack in oredictEntries) {
            if (foundMatch)
                break
            val item = itemStack.item
            if (item is ItemBlock) {
                if (item.block === block)
                    foundMatch = true
            }
        }
        return foundMatch
    }

}
